### ISSUE TITLE
Pasting text in Entry does not update cursor position display

### ISSUE THREAD

**Issue by matwachich** [fyne-io/fyne#4181]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When pasting some text in an Entry widget, the caret/cursor position is not update.
It is always shown at the begining of the pasted text.

It's a display bug only, since when, after the paste, we type some text, it is inserted at the right position (after the pasted text).

### How to reproduce

Paste text in an Entry widget.
See the cursor position (wrong).
Type some text, the cursor is updated to the right position.

### Screenshots

Typing some text
![image](https://github.com/fyne-io/fyne/assets/1791230/be420fbd-4f62-4754-81a1-cb495063fe2a)

After paste
![image](https://github.com/fyne-io/fyne/assets/1791230/5d58794f-8fd4-4d94-b9fc-52a6a4483a9b)

Re-type some text (cursor is updated to the right position)
![image](https://github.com/fyne-io/fyne/assets/1791230/f7be2abd-99b9-4665-8183-4cd6b012fcaa)

### Example code

```go
package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("")

	w.SetContent(widget.NewMultiLineEntry())
	w.ShowAndRun()
}
```

### Fyne version

v2.3.6-0.20230822222624-fce948601c72

### Go compiler version

1.21

### Operating system and version

Windows 10

**Comment by Jacalz** [comment]
This is a regression for v2.4.0. Adding to the fixes milestone

**Comment by andydotxyz** [comment]
Fixed on develop and in release v2.4.2
